test/cases/ref.ts

```diff
@@ -0,0 +1,40 @@
+export var schema = 
+{
+  "title": "Referencing",
+  "type": "object",
+  "properties": {
+    "foo": {
+      "$ref": "test/resources/ReferencedType.json"
+    }
+  },
+  "required": ["foo"],
+  "additionalProperties": false
+}
+
+export var configurations = [
+  {
+    settings: {
+      declareReferenced: true
```

Comment:
If >1 declaration is generated, they will be merged by TSC with no overlap. I think a better option would be to add a method to json-schema-to-typescript to compile multiple schemas at once; that way we let it manage deduping for us.